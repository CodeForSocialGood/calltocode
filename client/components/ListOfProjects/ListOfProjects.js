import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import GridList from 'material-ui/GridList'

import UserActionCreator from '../../actions/user'
import Project from '../Project/Project'
import styles from './ListOfProjects.scss'

class ListOfProjects extends Component {
  constructor (props) {
    super(props)

    this.renderListOfProjects = this.renderListOfProjects.bind(this)
  }

  renderListOfProjects () {
    return this.props.projects.map((project, index) => {
      return (
        <Project key={index}
          project={project}
          authenticated={this.props.authenticated}
          applyForProject={this.props.applyForProject} />
      )
    })
  }

  render () {
    return (
      <section className={styles.projectListSection}>
        <h1 className={styles.title}>{this.props.title}</h1>

        <div className={styles.listContainer}>
          <GridList className={styles.list} cols={3}>
            { this.renderListOfProjects() }
          </GridList>
        </div>
      </section>
    )
  }
}

function mapStateToProps (state) {
  return {
    authenticated: state.auth.authenticated
  }
}

const mapDispatchToProps = {
  applyForProject: UserActionCreator.applyForProject
}

ListOfProjects.propTypes = {
  applyForProject: PropTypes.func,
  authenticated: PropTypes.bool.isRequired,
  projects: PropTypes.array,
  title: PropTypes.string
}

export default connect(mapStateToProps, mapDispatchToProps)(ListOfProjects)
