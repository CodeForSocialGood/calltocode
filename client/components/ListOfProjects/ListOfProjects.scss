@import '../../scss/partials/partials';

.projectListSection {
  padding: 0 60px;
  display: grid;
}

.listContainer {
  width: 100% !important;
  display: flex !important;
  flex-wrap: wrap !important;
}

.list {
  width: 100% !important;
}

.projectTile {
    margin-bottom: 20px;
}

.causesContainer {
  display: inline-flex;
  width: 100%;
  overflow-x: scroll;
}

.cause {
  margin-bottom: 70px;
  margin-right: 5px;
}

.projectTile {
  display: flex;
}
