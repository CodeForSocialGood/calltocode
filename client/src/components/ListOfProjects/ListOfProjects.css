.title {
  font-size: 20px;
  margin-bottom: 15px;
}

.orgSection {
  display: block;
  padding: 0 60px;
  margin-top: 30px;
}

.listOrg    {
  padding: 20px;
  border: 0;
  border-radius: 5px;
  color: white;
  background-color: darkgray;
  margin-bottom: 15px;
}

li:hover{
  box-shadow: 0 6px 4px -2px #808080;
  cursor: pointer;
}
