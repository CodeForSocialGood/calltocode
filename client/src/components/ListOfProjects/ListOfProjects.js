import styles from './ListOfProjects.css'
import React from 'react'
import projects from '../../data/projects.json'
import emailApiClient from '../../apiClients/email'

function ListOfProjects () {
  const projectListItems = projects.map((project, index) => {
    return <li key={index} className={styles.listOrg} onClick={mailToOrganization(project)}>Name:{project.name} Role:{project.role}</li>
  })

  return (
    <section className={styles.orgSection}>
      <h1 className={styles.title}>Apply Below</h1>
      <ul>
        {projectListItems}
      </ul>
    </section>
  )
}

function mailToOrganization (project) {
  return () => {
    const {email, name, role} = project
    const projectInfo = {email, name, role}

    emailApiClient.send(projectInfo)
  }
}

export default ListOfProjects
