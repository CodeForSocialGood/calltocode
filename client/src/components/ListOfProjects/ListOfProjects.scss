@import '../../scss/partials/partials';

.title {
  font-size: 20px;
  margin-bottom: 15px;
}

.orgSection {
  display: block;
  padding: 0 60px;
  margin-top: 30px;
}

.listOrg {
  padding: 20px;
  border: 0;
  border-radius: 5px;
  color: white;
  background-color: $listColor;
  margin-bottom: 15px;
}

.listOrgLoggedIn {
  @extend .listOrg;
}

.listOrgLoggedIn:hover {
  @extend %hover;
  @extend %boxShadowSmall;
}
