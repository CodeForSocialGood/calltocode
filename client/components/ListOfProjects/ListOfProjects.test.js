/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import {ListOfProjects} from './ListOfProjects'
let mockProps

beforeEach(() => {
  mockProps = {
    applyForProject: () => {},
    authenticated: false,
    projects: [],
    title: ''
  }
})

describe('<ListOfProjects />', () => {
  describe('list of projects is empty', () => {
    describe('autheticated is truthy', () => {
      beforeEach(() => { mockProps.authenticated = true })

      test('component renders correctly', () => {
        const component = shallow(<ListOfProjects {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })
    describe('authenticated is falsey', () => {
      beforeEach(() => { mockProps.authenticated = false })

      test('component renders correctly', () => {
        const component = shallow(<ListOfProjects {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })
  })
  describe('list of projects is not empty', () => {
    beforeEach(() => { mockProps.projects = [{}] })

    describe('autheticated is truthy', () => {
      beforeEach(() => { mockProps.authenticated = true })

      test('component renders correctly', () => {
        const component = shallow(<ListOfProjects {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })
    describe('authenticated is falsey', () => {
      beforeEach(() => { mockProps.authenticated = false })

      test('component renders correctly', () => {
        const component = shallow(<ListOfProjects {...mockProps} />)
        expect(component).toMatchSnapshot()
      })
    })
  })
})
