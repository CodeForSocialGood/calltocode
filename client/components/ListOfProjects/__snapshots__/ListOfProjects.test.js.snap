// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfProjects /> list of projects is empty authenticated is falsey component renders correctly 1`] = `
<section
  className="projectListSection"
>
  <h1
    className="title"
  />
  <div
    className="listContainer"
  >
    <WithStyles(GridList)
      className="list"
      cols={3}
    />
  </div>
</section>
`;

exports[`<ListOfProjects /> list of projects is empty autheticated is truthy component renders correctly 1`] = `
<section
  className="projectListSection"
>
  <h1
    className="title"
  />
  <div
    className="listContainer"
  >
    <WithStyles(GridList)
      className="list"
      cols={3}
    />
  </div>
</section>
`;

exports[`<ListOfProjects /> list of projects is not empty authenticated is falsey component renders correctly 1`] = `
<section
  className="projectListSection"
>
  <h1
    className="title"
  />
  <div
    className="listContainer"
  >
    <WithStyles(GridList)
      className="list"
      cols={3}
    >
      <Connect(Project)
        applyForProject={[Function]}
        authenticated={false}
        key="0"
        project={Object {}}
      />
    </WithStyles(GridList)>
  </div>
</section>
`;

exports[`<ListOfProjects /> list of projects is not empty autheticated is truthy component renders correctly 1`] = `
<section
  className="projectListSection"
>
  <h1
    className="title"
  />
  <div
    className="listContainer"
  >
    <WithStyles(GridList)
      className="list"
      cols={3}
    >
      <Connect(Project)
        applyForProject={[Function]}
        authenticated={true}
        key="0"
        project={Object {}}
      />
    </WithStyles(GridList)>
  </div>
</section>
`;
